lib/AST/Decl.cpp

```diff
@@ -5246,6 +5246,18 @@ static bool requiresNewVTableEntry(const AbstractFunctionDecl *decl) {
   // Dynamic methods are always accessed by objc_msgSend().
   if (decl->isFinal() || decl->isDynamic() || decl->hasClangNode())
     return false;
+  
+  // Initializers are not normally inherited, but required initializers can
```

Comment:
I'm confused by this "inherited". Requiring a new vtable entry also happens when the type changes.

Reply:
A convenience initializer is always effectively `final`, though, so even if it overrides a designated init while changing type, it'll override an existing vtable entry, but never need a new one of its own.

Reply:
Ah, sure, but this comment is about all initializers. I would have expected something like "Only designated initializers and `required` convenience initializers appear in the vtable; bail out if this is an initializer that is neither designated nor `required`."